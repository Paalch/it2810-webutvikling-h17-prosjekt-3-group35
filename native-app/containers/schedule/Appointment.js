import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';

export default class Appointment extends React.Component {
    render() {
        return (
            <View className={'appointment-container'} style={{backgroundColor: this.props.color}}>
                <label className={"appointment-content"}>{this.props.time}</label>
                <label>{this.props.text}</label>
                <Button
                    className={"delete-appointment"}
                    onPress={() => this.props.deleteAppointment(this.props.appointment.id)}
                    style={{backgroundColor: this.props.color}}
                    Title="X"/>

            </View>
        );
    }
}
