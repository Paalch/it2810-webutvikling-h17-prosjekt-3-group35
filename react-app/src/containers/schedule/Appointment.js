import React, {Component} from 'react'
import {Button} from 'react-bootstrap'
import "./schedule.css"
import moment from 'moment'


class Appointment extends Component {

  constructor(props) {
    super(props);
    this.state = {
      appointments: this.props.appointments
    };
    this.deleteAppointment = this.deleteAppointment.bind(this);
  }

  deleteAppointment() {
    let appointments = this.state.appointments;
    for (let i = 0; i < appointments.length; i++) {
      if (appointments[i].includes(this.props.id)) {
        appointments.splice(i, 1);
      }
    }
    this.setState({
      appointments: appointments
    });

    this.props.receiveAppointment(this.state.appointments);
  }

    interval = setInterval(() => {
      if (this.state.appointments.length > 0){
          let appointments = this.state.appointments;
          for (let i = 0; i < appointments.length; i++) {
              if(appointments[i][2].format("DD.MM.YYYY HH:mm") < moment().subtract(1, "hours").format("MM.DD.YYYY HH:mm")){
                  appointments.splice(i, 1)
              }
          }
          this.setState({
              appointments: appointments
          });

          this.props.receiveAppointment(this.state.appointments);
      }
    }, 1000);



  render() {
    return (
      <div className={'appointment-container'}>
            <label className={"appointment-content"}>{this.props.time}</label>
            <label>{this.props.text}</label>
            <Button className={"delete-appointment"} onClick={this.deleteAppointment}>X</Button>
      </div>
    );
  }
}

export default Appointment;