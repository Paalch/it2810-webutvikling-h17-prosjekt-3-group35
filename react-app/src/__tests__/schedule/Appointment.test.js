import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import renderer from 'react-test-renderer'
import Appointment from "../../containers/schedule/Appointment";


Enzyme.configure({ adapter: new Adapter() })


describe('Checks that the schedule component renders correctly', () => {

    it('Checks that Appointment renders', () => {
        const rendered = renderer.create(
            <Appointment />
        );
        expect(rendered.toJSON()).toMatchSnapshot()
    });

});
