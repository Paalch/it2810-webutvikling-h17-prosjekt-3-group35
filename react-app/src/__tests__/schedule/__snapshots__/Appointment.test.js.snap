// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that the schedule component renders correctly Checks that Appointment renders 1`] = `
<div
  className="appointment-container"
>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-heading"
    >
      <div
        className="panel-title"
      >
        <div
          className="row"
        >
          <div
            className="col-xs-8"
          >
            <p />
          </div>
          <div
            className="col-xs-4"
          >
            <button
              className="delete-appointment btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              X
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
